### PR TITLE
support environment variables.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs [hunk 1]
```diff
@@ -308,11 +309,16 @@ public void InitFileAttributes()
                     FileAttributeItem item;
                     if (!TryGetFileAttributeFromLast(key, out item))
                     {
+                        string md5;
+                        using (var fs = File.OpenRead(f.FullPath))
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs]: >OpenRead [](start = 45, length = 8)

q: why update here?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs]: This is a defect, file may be in binary format.
Move to another PR.

---
In reply to: [106141309](https://github.com/dotnet/docfx/pull/1417#discussion_r106141309) [](ancestors = 106141309)

**Comment by hellosnow** [comment]
:shipit:
